upstash_vector/core/resumable_query.py

```diff
@@ -0,0 +1,100 @@
+from typing import List, Optional, Dict, Any
+from upstash_vector.types import QueryResult
+from upstash_vector.errors import ClientError
+
+RESUMABLE_QUERY_VECTOR_PATH = "/resumable-query"
+RESUMABLE_QUERY_DATA_PATH = "/resumable-query-data"
+RESUMABLE_QUERY_NEXT_PATH = "/resumable-query-next"
+RESUMABLE_QUERY_END_PATH = "/resumable-query-end"
+
+
+class ResumableQuery:
+    def __init__(
+        self, payload: Dict[str, Any], client, namespace: Optional[str] = None
+    ):
+        self.payload = payload
+        self.client = client
+        self.namespace = namespace
+        self.uuid = None
+
+    async def async_start(self) -> List[QueryResult]:
```

Comment:
Can we add __enter__ and __exit__ functions to support `with` statements. This pattern would really fit our usacase. It's optional of course, and we can add an example of it in the docs
https://www.geeksforgeeks.org/with-statement-in-python/

```py
    def __enter__(self):
        pass

    def __exit__(self, *args):
        self.stop()
```

Reply:
We will probably need to return `self` in `__enter__`, so that we can use fetch_next:

```py
with client.resumable_query(...) as query:
    next = query.fetch_next()
    # ...
```

Reply:
We will probably need to return `self` in `__enter__`, so that we can use fetch_next:

```py
with client.resumable_query(...) as query:
    next = query.fetch_next()
    # ...
```

Reply:
This is implemented now